### PR TITLE
Alert modal

### PR DESCRIPTION
**Pull request by crmorford**

### CODE REVIEW

**Comment by crmorford** [comment]
So, i did this with two components/templates/js files at first -- then realized i could combine the components + js files. Now thinking I should probably combine the templates too and just pass in a variable.

**Comment by ilyakamens** [comment]
image

missing border radius

File: examples/index.html [hunk 1]
```diff
@@ -31,10 +32,17 @@
       <div class="mp-modal-subtitle" slot-body>Click on any element on your page (ex. the buy button) to create an event and access all historical data for that event.</div>
       <mp-button class="modal" onclick="closePopup();" slot-button>Create your first event</mp-button>
     </mp-modal>
+    <mp-modal id="alertModal" modal-type="popup" style="display: none" closeable alert>
+      <div slot-body>
+        <div class="alert-modal-title">Your event failed to save</div>
+        <div class="alert-modal-subtitle"">If your issue persists please contact our support team at support@mixpanel.com to get help quickly</div>
```
Comment (ilyakamens) [inline examples/index.html]: looks like you've got an extra double quote here

File: examples/index.html [hunk 2]
```diff
@@ -31,10 +32,17 @@
       <div class="mp-modal-subtitle" slot-body>Click on any element on your page (ex. the buy button) to create an event and access all historical data for that event.</div>
       <mp-button class="modal" onclick="closePopup();" slot-button>Create your first event</mp-button>
     </mp-modal>
+    <mp-modal id="alertModal" modal-type="popup" style="display: none" closeable alert>
```
Comment (ilyakamens) [inline examples/index.html]: minor, but in general our style is to use hyphenated values for ids, classes, etc. (i.e. `alert-modal`)

File: lib/components/modals/modal.jade [hunk 3]
```diff
@@ -1,4 +1,4 @@
-.mp-modal
+.mp-modal(class=alertIcon?"alert":"")
```
Comment (ilyakamens) [inline lib/components/modals/modal.jade]: missing some spacing here

File: lib/components/modals/modal.js [hunk 4]
```diff
@@ -8,6 +8,9 @@ document.registerElement('mp-modal', class extends Component {
   get config() {
     return {
       css,
+      defaultState: {
+        alertIcon: this.isAttributeEnabled('alert'),
```
Comment (ilyakamens) [inline lib/components/modals/modal.js]: can we add a verb here like `hasAlertIcon`

File: lib/components/modals/modal.js [hunk 5]
```diff
@@ -62,6 +65,9 @@ document.registerElement('mp-modal', class extends Component {
 
   attachedCallback() {
     super.attachedCallback(...arguments);
+    if (this.isAttributeEnabled('alert')) {
+      this.el.querySelector('.mp-modal').classList.add('alert');
```
Comment (ilyakamens) [inline lib/components/modals/modal.js]: i know this pattern is used immediately below, but i don't think you need this. shouldn't the `alert` class be added to `.mp-modal` in the jade template already?

File: lib/components/modals/modal.styl [hunk 6]
```diff
@@ -109,3 +112,33 @@
...
+      margin-right 15px
+      white-space normal
+      width 33px
+
+      svg 
+        height 29px
+        .fill
+         fill mp-red
+
+
```
Comment (ilyakamens) [inline lib/components/modals/modal.styl]: extra line. if you add `"ensure_newline_at_eof_on_save": true,` to your Sublime preferences (`command + comma` when in the app) it should ensure there's always one newline at the end of the file

**Comment by ilyakamens** [comment]
LGTM%C!

File: lib/components/modals/modal.js (hunk 5 above)
Comment (tdumitrescu) [inline lib/components/modals/modal.js]: yes, manipulating the DOM imperatively is pretty much the #1 anti-pattern in virtual DOM style. This needs to be in a template

Comment (jbwyme) [inline lib/components/modals/modal.js]: except animations.....

Comment (tdumitrescu) [inline lib/components/modals/modal.js]: if the idea is that this class gets added after the template has been rendered, then this code will only work on chrome; on other browsers it won't get rendered til the next animation frame. I still 100% think this needs to be done with state updates, not direct dom manipulation.

Comment (jbwyme) [inline lib/components/modals/modal.js]: not disagreeing, just pointing out that animations fuck us on not manipulating the dom imperatively (or through state as a straight up proxy). cassie you can ignore my comments here.

Comment (tdumitrescu) [inline lib/components/modals/modal.js]: State as a proxy for imperative DOM changes might not be philosophically perfect, but it's at least guaranteed to work. Messing directly with elements added by the virtual dom renderer is a recipe for undefined behavior.

**Pull request merged**